### PR TITLE
ytdl-sub: init at 2024.10.26

### PR DESCRIPTION
**Pull request by LoCrealloc**
https://github.com/jmbannon/ytdl-sub

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/yt/ytdl-sub/package.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    yt-dlp
+    colorama
+    mergedeep
+    mediafile
+    pyyaml
+  ];
+
+  makeWrapperArgs = [
+    "--set YTDL_SUB_FFMPEG_PATH ${ffmpeg}/bin/ffmpeg"
+    "--set YTDL_SUB_FFPROBE_PATH ${ffmpeg}/bin/ffprobe"
```
Comment (DontEatOreo) [inline pkgs/by-name/yt/ytdl-sub/package.nix:33]:
```suggestion
    "--set YTDL_SUB_FFMPEG_PATH ${lib.getExe' ffmpeg "ffmpeg"}"
    "--set YTDL_SUB_FFPROBE_PATH ${lib.getExe' ffmpeg "ffprobe"}"
```

**Review by 4evy — CHANGES_REQUESTED** [review]

File: pkgs/by-name/yt/ytdl-sub/package.nix (hunk 1 above)
Comment (LoCrealloc) [reply pkgs/by-name/yt/ytdl-sub/package.nix]: thanks!

File: pkgs/by-name/yt/ytdl-sub/package.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  meta = with lib; {
+    homepage = "https://github.com/jmbannon/ytdl-sub";
+    description = "Lightweight tool to automate downloading and metadata generation with yt-dlp ";
+    longDescription = ''
+      ytdl-sub is a command-line tool that downloads media via yt-dlp and prepares it for your favorite media player, including Kodi, Jellyfin, Plex, Emby, and modern music players. No additional plugins or external scrapers are needed.
+    '';
+    changelog = "https://github.com/jmbannon/ytdl-sub/releases/tag/${version}";
+    license = licenses.gpl3;
+    maintainers = with maintainers; [
```
Comment (Aleksanaa) [inline pkgs/by-name/yt/ytdl-sub/package.nix:44]:
```suggestion
  meta = {
    homepage = "https://github.com/jmbannon/ytdl-sub";
    description = "Lightweight tool to automate downloading and metadata generation with yt-dlp ";
    longDescription = ''
      ytdl-sub is a command-line tool that downloads media via yt-dlp and prepares it for your favorite media player, including Kodi, Jellyfin, Plex, Emby, and modern music players. No additional plugins or external scrapers are needed.
    '';
    changelog = "https://github.com/jmbannon/ytdl-sub/releases/tag/${version}";
    license = lib.licenses.gpl3Only;
    maintainers = with lib.maintainers; [
```

Comment (LoCrealloc) [reply pkgs/by-name/yt/ytdl-sub/package.nix]: thank you!

File: pkgs/by-name/yt/ytdl-sub/package.nix [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  ];
+
+  makeWrapperArgs = [
+    "--set YTDL_SUB_FFMPEG_PATH ${lib.getExe' ffmpeg "ffmpeg"}"
+    "--set YTDL_SUB_FFPROBE_PATH ${lib.getExe' ffmpeg "ffprobe"}"
+  ];
+
+  meta = {
+    homepage = "https://github.com/jmbannon/ytdl-sub";
+    description = "Lightweight tool to automate downloading and metadata generation with yt-dlp ";
```
Comment (Aleksanaa) [inline pkgs/by-name/yt/ytdl-sub/package.nix:38]:
```suggestion
    description = "Lightweight tool to automate downloading and metadata generation with yt-dlp";
```

**Review by 4evy — APPROVED** [review]
LGTM

**Review by Defelo — APPROVED** [review]
Built and tested successfully on `x86_64-linux` and `aarch64-linux`

**Pull request merged**